From: Stained Glass OS <ke7oxh@gmail.com>
Subject: [PATCH] msvcp70: Export the functions that match msvcp90's by signature in the old msvcp versions.

Fast mode for msvcp60 .. msvcp80: spec stubs whose demangled signature
equals that of a function msvcp90 already exports, now naming that
function (about 200 entries, mostly msvcp70's complex functions and a few
stream operators for signed/unsigned char strings). Matched with
scripts/demangle_map.py (winedump's demangler); entries whose function the
DLL does not build (msvcp60's own ios.c) stay stubs.
---
 dlls/msvcp60/msvcp60.spec |  28 ++--
 dlls/msvcp70/msvcp70.spec | 340 +++++++++++++++++++++++-----------------------
 dlls/msvcp71/msvcp71.spec |   8 +-
 dlls/msvcp80/msvcp80.spec |   4 +-
 4 files changed, 190 insertions(+), 190 deletions(-)

diff --git a/dlls/msvcp60/msvcp60.spec b/dlls/msvcp60/msvcp60.spec
index b192943..830c32e 100644
--- a/dlls/msvcp60/msvcp60.spec
+++ b/dlls/msvcp60/msvcp60.spec
@@ -8,8 +8,8 @@
 @ cdecl -arch=win64 ??$?5DU?$char_traits@D@std@@@std@@YAAEAV?$basic_istream@DU?$char_traits@D@std@@@0@AEAV10@PEAC@Z(ptr ptr) basic_istream_char_read_ch
 @ cdecl -arch=win32 ??$?5DU?$char_traits@D@std@@@std@@YAAAV?$basic_istream@DU?$char_traits@D@std@@@0@AAV10@PAD@Z(ptr ptr) basic_istream_char_read_str
 @ cdecl -arch=win64 ??$?5DU?$char_traits@D@std@@@std@@YAAEAV?$basic_istream@DU?$char_traits@D@std@@@0@AEAV10@PEAD@Z(ptr ptr) basic_istream_char_read_str
-@ stub -arch=win32 ??$?5DU?$char_traits@D@std@@@std@@YAAAV?$basic_istream@DU?$char_traits@D@std@@@0@AAV10@PAE@Z
-@ stub -arch=win64 ??$?5DU?$char_traits@D@std@@@std@@YAAEAV?$basic_istream@DU?$char_traits@D@std@@@0@AEAV10@PEAE@Z
+@ cdecl -arch=win32 ??$?5DU?$char_traits@D@std@@@std@@YAAAV?$basic_istream@DU?$char_traits@D@std@@@0@AAV10@PAE@Z(ptr ptr) basic_istream_char_read_str
+@ cdecl -arch=win64 ??$?5DU?$char_traits@D@std@@@std@@YAAEAV?$basic_istream@DU?$char_traits@D@std@@@0@AEAV10@PEAE@Z(ptr ptr) basic_istream_char_read_str
 @ stub -arch=win32 ??$?5DU?$char_traits@D@std@@M@std@@YAAAV?$basic_istream@DU?$char_traits@D@std@@@0@AAV10@AAV?$complex@M@0@@Z
 @ stub -arch=win64 ??$?5DU?$char_traits@D@std@@M@std@@YAAEAV?$basic_istream@DU?$char_traits@D@std@@@0@AEAV10@AEAV?$complex@M@0@@Z
 @ stub -arch=win32 ??$?5DU?$char_traits@D@std@@N@std@@YAAAV?$basic_istream@DU?$char_traits@D@std@@@0@AAV10@AAV?$complex@N@0@@Z
@@ -32,18 +32,18 @@
 @ stub -arch=win64 ??$?5GU?$char_traits@G@std@@O@std@@YAAEAV?$basic_istream@GU?$char_traits@G@std@@@0@AEAV10@AEAV?$complex@O@0@@Z
 @ cdecl -arch=win32 ??$?5GU?$char_traits@G@std@@V?$allocator@G@1@@std@@YAAAV?$basic_istream@GU?$char_traits@G@std@@@0@AAV10@AAV?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@0@@Z(ptr ptr) basic_istream_short_read_bstr
 @ cdecl -arch=win64 ??$?5GU?$char_traits@G@std@@V?$allocator@G@1@@std@@YAAEAV?$basic_istream@GU?$char_traits@G@std@@@0@AEAV10@AEAV?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@0@@Z(ptr ptr) basic_istream_short_read_bstr
-@ stub -arch=win32 ??$?6DU?$char_traits@D@std@@@std@@YAAAV?$basic_ostream@DU?$char_traits@D@std@@@0@AAV10@C@Z
-@ stub -arch=win64 ??$?6DU?$char_traits@D@std@@@std@@YAAEAV?$basic_ostream@DU?$char_traits@D@std@@@0@AEAV10@C@Z
-@ stub -arch=win32 ??$?6DU?$char_traits@D@std@@@std@@YAAAV?$basic_ostream@DU?$char_traits@D@std@@@0@AAV10@D@Z
-@ stub -arch=win64 ??$?6DU?$char_traits@D@std@@@std@@YAAEAV?$basic_ostream@DU?$char_traits@D@std@@@0@AEAV10@D@Z
-@ stub -arch=win32 ??$?6DU?$char_traits@D@std@@@std@@YAAAV?$basic_ostream@DU?$char_traits@D@std@@@0@AAV10@E@Z
-@ stub -arch=win64 ??$?6DU?$char_traits@D@std@@@std@@YAAEAV?$basic_ostream@DU?$char_traits@D@std@@@0@AEAV10@E@Z
-@ stub -arch=win32 ??$?6DU?$char_traits@D@std@@@std@@YAAAV?$basic_ostream@DU?$char_traits@D@std@@@0@AAV10@PBC@Z
-@ stub -arch=win64 ??$?6DU?$char_traits@D@std@@@std@@YAAEAV?$basic_ostream@DU?$char_traits@D@std@@@0@AEAV10@PEBC@Z
-@ stub -arch=win32 ??$?6DU?$char_traits@D@std@@@std@@YAAAV?$basic_ostream@DU?$char_traits@D@std@@@0@AAV10@PBD@Z
-@ stub -arch=win64 ??$?6DU?$char_traits@D@std@@@std@@YAAEAV?$basic_ostream@DU?$char_traits@D@std@@@0@AEAV10@PEBD@Z
-@ stub -arch=win32 ??$?6DU?$char_traits@D@std@@@std@@YAAAV?$basic_ostream@DU?$char_traits@D@std@@@0@AAV10@PBE@Z
-@ stub -arch=win64 ??$?6DU?$char_traits@D@std@@@std@@YAAEAV?$basic_ostream@DU?$char_traits@D@std@@@0@AEAV10@PEBE@Z
+@ cdecl -arch=win32 ??$?6DU?$char_traits@D@std@@@std@@YAAAV?$basic_ostream@DU?$char_traits@D@std@@@0@AAV10@C@Z(ptr long) basic_ostream_char_print_ch
+@ cdecl -arch=win64 ??$?6DU?$char_traits@D@std@@@std@@YAAEAV?$basic_ostream@DU?$char_traits@D@std@@@0@AEAV10@C@Z(ptr long) basic_ostream_char_print_ch
+@ cdecl -arch=win32 ??$?6DU?$char_traits@D@std@@@std@@YAAAV?$basic_ostream@DU?$char_traits@D@std@@@0@AAV10@D@Z(ptr long) basic_ostream_char_print_ch
+@ cdecl -arch=win64 ??$?6DU?$char_traits@D@std@@@std@@YAAEAV?$basic_ostream@DU?$char_traits@D@std@@@0@AEAV10@D@Z(ptr long) basic_ostream_char_print_ch
+@ cdecl -arch=win32 ??$?6DU?$char_traits@D@std@@@std@@YAAAV?$basic_ostream@DU?$char_traits@D@std@@@0@AAV10@E@Z(ptr long) basic_ostream_char_print_ch
+@ cdecl -arch=win64 ??$?6DU?$char_traits@D@std@@@std@@YAAEAV?$basic_ostream@DU?$char_traits@D@std@@@0@AEAV10@E@Z(ptr long) basic_ostream_char_print_ch
+@ cdecl -arch=win32 ??$?6DU?$char_traits@D@std@@@std@@YAAAV?$basic_ostream@DU?$char_traits@D@std@@@0@AAV10@PBC@Z(ptr ptr) basic_ostream_char_print_str
+@ cdecl -arch=win64 ??$?6DU?$char_traits@D@std@@@std@@YAAEAV?$basic_ostream@DU?$char_traits@D@std@@@0@AEAV10@PEBC@Z(ptr ptr) basic_ostream_char_print_str
+@ cdecl -arch=win32 ??$?6DU?$char_traits@D@std@@@std@@YAAAV?$basic_ostream@DU?$char_traits@D@std@@@0@AAV10@PBD@Z(ptr str) basic_ostream_char_print_str
+@ cdecl -arch=win64 ??$?6DU?$char_traits@D@std@@@std@@YAAEAV?$basic_ostream@DU?$char_traits@D@std@@@0@AEAV10@PEBD@Z(ptr str) basic_ostream_char_print_str
+@ cdecl -arch=win32 ??$?6DU?$char_traits@D@std@@@std@@YAAAV?$basic_ostream@DU?$char_traits@D@std@@@0@AAV10@PBE@Z(ptr ptr) basic_ostream_char_print_str
+@ cdecl -arch=win64 ??$?6DU?$char_traits@D@std@@@std@@YAAEAV?$basic_ostream@DU?$char_traits@D@std@@@0@AEAV10@PEBE@Z(ptr ptr) basic_ostream_char_print_str
 @ cdecl -arch=win32 ??$?6DU?$char_traits@D@std@@M@std@@YAAAV?$basic_ostream@DU?$char_traits@D@std@@@0@AAV10@ABV?$complex@M@0@@Z(ptr ptr) basic_ostream_char_print_complex_float
 @ cdecl -arch=win64 ??$?6DU?$char_traits@D@std@@M@std@@YAAEAV?$basic_ostream@DU?$char_traits@D@std@@@0@AEAV10@AEBV?$complex@M@0@@Z(ptr ptr) basic_ostream_char_print_complex_float
 @ cdecl -arch=win32 ??$?6DU?$char_traits@D@std@@N@std@@YAAAV?$basic_ostream@DU?$char_traits@D@std@@@0@AAV10@ABV?$complex@N@0@@Z(ptr ptr) basic_ostream_char_print_complex_double
diff --git a/dlls/msvcp70/msvcp70.spec b/dlls/msvcp70/msvcp70.spec
index 07341e3..160c430 100644
--- a/dlls/msvcp70/msvcp70.spec
+++ b/dlls/msvcp70/msvcp70.spec
@@ -1,12 +1,12 @@
 @ stub -arch=arm ??0?$_Complex_base@M@std@@QAA@ABM0@Z
-@ stub -arch=i386 ??0?$_Complex_base@M@std@@QAE@ABM0@Z
-@ stub -arch=win64 ??0?$_Complex_base@M@std@@QEAA@AEBM0@Z
+@ thiscall -arch=i386 ??0?$_Complex_base@M@std@@QAE@ABM0@Z(ptr ptr ptr) complex_float_ctor
+@ cdecl -arch=win64 ??0?$_Complex_base@M@std@@QEAA@AEBM0@Z(ptr ptr ptr) complex_float_ctor
 @ stub -arch=arm ??0?$_Complex_base@N@std@@QAA@ABN0@Z
-@ stub -arch=i386 ??0?$_Complex_base@N@std@@QAE@ABN0@Z
-@ stub -arch=win64 ??0?$_Complex_base@N@std@@QEAA@AEBN0@Z
+@ thiscall -arch=i386 ??0?$_Complex_base@N@std@@QAE@ABN0@Z(ptr ptr ptr) complex_double_ctor
+@ cdecl -arch=win64 ??0?$_Complex_base@N@std@@QEAA@AEBN0@Z(ptr ptr ptr) complex_double_ctor
 @ stub -arch=arm ??0?$_Complex_base@O@std@@QAA@ABO0@Z
-@ stub -arch=i386 ??0?$_Complex_base@O@std@@QAE@ABO0@Z
-@ stub -arch=win64 ??0?$_Complex_base@O@std@@QEAA@AEBO0@Z
+@ thiscall -arch=i386 ??0?$_Complex_base@O@std@@QAE@ABO0@Z(ptr ptr ptr) complex_double_ctor
+@ cdecl -arch=win64 ??0?$_Complex_base@O@std@@QEAA@AEBO0@Z(ptr ptr ptr) complex_double_ctor
 @ cdecl -arch=arm ??0?$_Mpunct@D@std@@QAA@ABV_Locinfo@1@I_N@Z(ptr long long) mpunct_char_ctor_refs
 @ thiscall -arch=i386 ??0?$_Mpunct@D@std@@QAE@ABV_Locinfo@1@I_N@Z(ptr long long) mpunct_char_ctor_refs
 @ cdecl -arch=win64 ??0?$_Mpunct@D@std@@QEAA@AEBV_Locinfo@1@_K_N@Z(ptr long long) mpunct_char_ctor_refs
@@ -1042,14 +1042,14 @@
 @ thiscall -arch=i386 ??1time_base@std@@UAE@XZ(ptr) time_base_dtor
 @ cdecl -arch=win64 ??1time_base@std@@UEAA@XZ(ptr) time_base_dtor
 @ stub -arch=arm ??4?$_Complex_base@M@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4?$_Complex_base@M@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4?$_Complex_base@M@std@@QEAAAEAV01@AEBV01@@Z
+@ thiscall -arch=i386 ??4?$_Complex_base@M@std@@QAEAAV01@ABV01@@Z(ptr ptr) complex_float_assign
+@ cdecl -arch=win64 ??4?$_Complex_base@M@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) complex_float_assign
 @ stub -arch=arm ??4?$_Complex_base@N@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4?$_Complex_base@N@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4?$_Complex_base@N@std@@QEAAAEAV01@AEBV01@@Z
+@ thiscall -arch=i386 ??4?$_Complex_base@N@std@@QAEAAV01@ABV01@@Z(ptr ptr) complex_double_assign
+@ cdecl -arch=win64 ??4?$_Complex_base@N@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) complex_double_assign
 @ stub -arch=arm ??4?$_Complex_base@O@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4?$_Complex_base@O@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4?$_Complex_base@O@std@@QEAAAEAV01@AEBV01@@Z
+@ thiscall -arch=i386 ??4?$_Complex_base@O@std@@QAEAAV01@ABV01@@Z(ptr ptr) complex_double_assign
+@ cdecl -arch=win64 ??4?$_Complex_base@O@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) complex_double_assign
 @ cdecl -arch=arm ??4?$_Ctraits@M@std@@QAAAAV01@ABV01@@Z(ptr ptr) std_Ctraits_op_assign
 @ thiscall -arch=i386 ??4?$_Ctraits@M@std@@QAEAAV01@ABV01@@Z(ptr ptr) std_Ctraits_op_assign
 @ cdecl -arch=win64 ??4?$_Ctraits@M@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) std_Ctraits_op_assign
@@ -1264,8 +1264,8 @@
 @ thiscall -arch=i386 ??5?$basic_istream@GU?$char_traits@G@std@@@std@@QAEAAV01@AAF@Z(ptr ptr) basic_istream_short_read_short
 @ cdecl -arch=win64 ??5?$basic_istream@GU?$char_traits@G@std@@@std@@QEAAAEAV01@AEAF@Z(ptr ptr) basic_istream_short_read_short
 @ stub -arch=arm ??5?$basic_istream@GU?$char_traits@G@std@@@std@@QAAAAV01@AAG@Z
-@ stub -arch=i386 ??5?$basic_istream@GU?$char_traits@G@std@@@std@@QAEAAV01@AAG@Z
-@ stub -arch=win64 ??5?$basic_istream@GU?$char_traits@G@std@@@std@@QEAAAEAV01@AEAG@Z
+@ thiscall -arch=i386 ??5?$basic_istream@GU?$char_traits@G@std@@@std@@QAEAAV01@AAG@Z(ptr ptr) basic_istream_wchar_read_ushort
+@ cdecl -arch=win64 ??5?$basic_istream@GU?$char_traits@G@std@@@std@@QEAAAEAV01@AEAG@Z(ptr ptr) basic_istream_wchar_read_ushort
 @ cdecl -arch=arm ??5?$basic_istream@GU?$char_traits@G@std@@@std@@QAAAAV01@AAH@Z(ptr ptr) basic_istream_short_read_int
 @ thiscall -arch=i386 ??5?$basic_istream@GU?$char_traits@G@std@@@std@@QAEAAV01@AAH@Z(ptr ptr) basic_istream_short_read_int
 @ cdecl -arch=win64 ??5?$basic_istream@GU?$char_traits@G@std@@@std@@QEAAAEAV01@AEAH@Z(ptr ptr) basic_istream_short_read_int
@@ -1370,12 +1370,12 @@
 @ cdecl -arch=win64 ??5std@@YAAEAV?$basic_istream@DU?$char_traits@D@std@@@0@AEAV10@AEAE@Z(ptr ptr) basic_istream_char_read_ch
 @ cdecl -arch=win32 ??5std@@YAAAV?$basic_istream@DU?$char_traits@D@std@@@0@AAV10@AAV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@0@@Z(ptr ptr) basic_istream_char_read_bstr
 @ cdecl -arch=win64 ??5std@@YAAEAV?$basic_istream@DU?$char_traits@D@std@@@0@AEAV10@AEAV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@0@@Z(ptr ptr) basic_istream_char_read_bstr
-@ stub -arch=win32 ??5std@@YAAAV?$basic_istream@DU?$char_traits@D@std@@@0@AAV10@AAV?$complex@M@0@@Z
-@ stub -arch=win64 ??5std@@YAAEAV?$basic_istream@DU?$char_traits@D@std@@@0@AEAV10@AEAV?$complex@M@0@@Z
-@ stub -arch=win32 ??5std@@YAAAV?$basic_istream@DU?$char_traits@D@std@@@0@AAV10@AAV?$complex@N@0@@Z
-@ stub -arch=win64 ??5std@@YAAEAV?$basic_istream@DU?$char_traits@D@std@@@0@AEAV10@AEAV?$complex@N@0@@Z
-@ stub -arch=win32 ??5std@@YAAAV?$basic_istream@DU?$char_traits@D@std@@@0@AAV10@AAV?$complex@O@0@@Z
-@ stub -arch=win64 ??5std@@YAAEAV?$basic_istream@DU?$char_traits@D@std@@@0@AEAV10@AEAV?$complex@O@0@@Z
+@ cdecl -arch=win32 ??5std@@YAAAV?$basic_istream@DU?$char_traits@D@std@@@0@AAV10@AAV?$complex@M@0@@Z(ptr ptr) basic_istream_char_read_complex_float
+@ cdecl -arch=win64 ??5std@@YAAEAV?$basic_istream@DU?$char_traits@D@std@@@0@AEAV10@AEAV?$complex@M@0@@Z(ptr ptr) basic_istream_char_read_complex_float
+@ cdecl -arch=win32 ??5std@@YAAAV?$basic_istream@DU?$char_traits@D@std@@@0@AAV10@AAV?$complex@N@0@@Z(ptr ptr) basic_istream_char_read_complex_double
+@ cdecl -arch=win64 ??5std@@YAAEAV?$basic_istream@DU?$char_traits@D@std@@@0@AEAV10@AEAV?$complex@N@0@@Z(ptr ptr) basic_istream_char_read_complex_double
+@ cdecl -arch=win32 ??5std@@YAAAV?$basic_istream@DU?$char_traits@D@std@@@0@AAV10@AAV?$complex@O@0@@Z(ptr ptr) basic_istream_char_read_complex_ldouble
+@ cdecl -arch=win64 ??5std@@YAAEAV?$basic_istream@DU?$char_traits@D@std@@@0@AEAV10@AEAV?$complex@O@0@@Z(ptr ptr) basic_istream_char_read_complex_ldouble
 @ cdecl -arch=win32 ??5std@@YAAAV?$basic_istream@DU?$char_traits@D@std@@@0@AAV10@PAC@Z(ptr ptr) basic_istream_char_read_str
 @ cdecl -arch=win64 ??5std@@YAAEAV?$basic_istream@DU?$char_traits@D@std@@@0@AEAV10@PEAC@Z(ptr ptr) basic_istream_char_read_str
 @ cdecl -arch=win32 ??5std@@YAAAV?$basic_istream@DU?$char_traits@D@std@@@0@AAV10@PAD@Z(ptr ptr) basic_istream_char_read_str
@@ -2428,12 +2428,12 @@
 @ cdecl -arch=win64 ?_Exp@?$_Ctraits@N@std@@SAFPEANNF@Z(ptr double long) _Exp
 @ cdecl -arch=win32 ?_Exp@?$_Ctraits@O@std@@SAFPAOOF@Z(ptr double long) _Exp
 @ cdecl -arch=win64 ?_Exp@?$_Ctraits@O@std@@SAFPEAOOF@Z(ptr double long) _Exp
-@ stub -arch=win32 ?_Fabs@std@@YAMABV?$complex@M@1@PAH@Z
-@ stub -arch=win64 ?_Fabs@std@@YAMAEBV?$complex@M@1@PEAH@Z
-@ stub -arch=win32 ?_Fabs@std@@YANABV?$complex@N@1@PAH@Z
-@ stub -arch=win64 ?_Fabs@std@@YANAEBV?$complex@N@1@PEAH@Z
-@ stub -arch=win32 ?_Fabs@std@@YAOABV?$complex@O@1@PAH@Z
-@ stub -arch=win64 ?_Fabs@std@@YAOAEBV?$complex@O@1@PEAH@Z
+@ cdecl -arch=win32 ?_Fabs@std@@YAMABV?$complex@M@1@PAH@Z(ptr ptr) complex_float__Fabs
+@ cdecl -arch=win64 ?_Fabs@std@@YAMAEBV?$complex@M@1@PEAH@Z(ptr ptr) complex_float__Fabs
+@ cdecl -arch=win32 ?_Fabs@std@@YANABV?$complex@N@1@PAH@Z(ptr ptr) complex_double__Fabs
+@ cdecl -arch=win64 ?_Fabs@std@@YANAEBV?$complex@N@1@PEAH@Z(ptr ptr) complex_double__Fabs
+@ cdecl -arch=win32 ?_Fabs@std@@YAOABV?$complex@O@1@PAH@Z(ptr ptr) complex_double__Fabs
+@ cdecl -arch=win64 ?_Fabs@std@@YAOAEBV?$complex@O@1@PEAH@Z(ptr ptr) complex_double__Fabs
 @ stub -arch=win32 ?_Ffmt@?$num_put@DV?$ostreambuf_iterator@DU?$char_traits@D@std@@@std@@@std@@KAPADPADDH@Z
 @ stub -arch=win64 ?_Ffmt@?$num_put@DV?$ostreambuf_iterator@DU?$char_traits@D@std@@@std@@@std@@KAPEADPEADDH@Z
 @ stub -arch=win32 ?_Ffmt@?$num_put@GV?$ostreambuf_iterator@GU?$char_traits@G@std@@@std@@@std@@KAPADPADDH@Z
@@ -3033,12 +3033,12 @@
 @ thiscall -arch=i386 ?_Xlen@_String_base@std@@QBEXXZ(ptr) _String_base__Xlen
 @ cdecl -arch=arm ?_Xran@_String_base@std@@QBAXXZ(ptr) _String_base__Xran
 @ thiscall -arch=i386 ?_Xran@_String_base@std@@QBEXXZ(ptr) _String_base__Xran
-@ stub -arch=win32 ?abs@std@@YAMABV?$complex@M@1@@Z
-@ stub -arch=win64 ?abs@std@@YAMAEBV?$complex@M@1@@Z
-@ stub -arch=win32 ?abs@std@@YANABV?$complex@N@1@@Z
-@ stub -arch=win64 ?abs@std@@YANAEBV?$complex@N@1@@Z
-@ stub -arch=win32 ?abs@std@@YAOABV?$complex@O@1@@Z
-@ stub -arch=win64 ?abs@std@@YAOAEBV?$complex@O@1@@Z
+@ cdecl -arch=win32 ?abs@std@@YAMABV?$complex@M@1@@Z(ptr) complex_float_abs
+@ cdecl -arch=win64 ?abs@std@@YAMAEBV?$complex@M@1@@Z(ptr) complex_float_abs
+@ cdecl -arch=win32 ?abs@std@@YANABV?$complex@N@1@@Z(ptr) complex_double_abs
+@ cdecl -arch=win64 ?abs@std@@YANAEBV?$complex@N@1@@Z(ptr) complex_double_abs
+@ cdecl -arch=win32 ?abs@std@@YAOABV?$complex@O@1@@Z(ptr) complex_double_abs
+@ cdecl -arch=win64 ?abs@std@@YAOAEBV?$complex@O@1@@Z(ptr) complex_double_abs
 @ cdecl -arch=arm ?always_noconv@codecvt_base@std@@QBA_NXZ(ptr) codecvt_base_always_noconv
 @ thiscall -arch=i386 ?always_noconv@codecvt_base@std@@QBE_NXZ(ptr) codecvt_base_always_noconv
 @ cdecl -arch=win64 ?always_noconv@codecvt_base@std@@QEBA_NXZ(ptr) codecvt_base_always_noconv
@@ -3102,12 +3102,12 @@
 @ cdecl -arch=win64 ?append@?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@std@@QEAAAEAV12@_K_W@Z(ptr long long) MSVCP_basic_string_wchar_append_len_ch
 @ cdecl -arch=arm ?append@?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@std@@QAAAAV12@V?$_Ptrit@_WHPB_WAB_WPA_WAA_W@2@0@Z(ptr ptr ptr) basic_string_wchar_append_iter
 @ thiscall -arch=i386 ?append@?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@std@@QAEAAV12@V?$_Ptrit@_WHPB_WAB_WPA_WAA_W@2@0@Z(ptr ptr ptr) basic_string_wchar_append_iter
-@ stub -arch=win32 ?arg@std@@YAMABV?$complex@M@1@@Z
-@ stub -arch=win64 ?arg@std@@YAMAEBV?$complex@M@1@@Z
-@ stub -arch=win32 ?arg@std@@YANABV?$complex@N@1@@Z
-@ stub -arch=win64 ?arg@std@@YANAEBV?$complex@N@1@@Z
-@ stub -arch=win32 ?arg@std@@YAOABV?$complex@O@1@@Z
-@ stub -arch=win64 ?arg@std@@YAOAEBV?$complex@O@1@@Z
+@ cdecl -arch=win32 ?arg@std@@YAMABV?$complex@M@1@@Z(ptr) complex_float_arg
+@ cdecl -arch=win64 ?arg@std@@YAMAEBV?$complex@M@1@@Z(ptr) complex_float_arg
+@ cdecl -arch=win32 ?arg@std@@YANABV?$complex@N@1@@Z(ptr) complex_double_arg
+@ cdecl -arch=win64 ?arg@std@@YANAEBV?$complex@N@1@@Z(ptr) complex_double_arg
+@ cdecl -arch=win32 ?arg@std@@YAOABV?$complex@O@1@@Z(ptr) complex_double_arg
+@ cdecl -arch=win64 ?arg@std@@YAOAEBV?$complex@O@1@@Z(ptr) complex_double_arg
 @ cdecl -arch=arm ?assign@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@QAAAAV12@ABV12@@Z(ptr ptr) MSVCP_basic_string_char_assign
 @ thiscall -arch=i386 ?assign@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@QAEAAV12@ABV12@@Z(ptr ptr) MSVCP_basic_string_char_assign
 @ cdecl -arch=win64 ?assign@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@QEAAAEAV12@AEBV12@@Z(ptr ptr) MSVCP_basic_string_char_assign
@@ -3391,12 +3391,12 @@
 @ cdecl -arch=arm ?compare@?$collate@_W@std@@QBAHPB_W000@Z(ptr ptr ptr ptr ptr) collate_wchar_compare
 @ thiscall -arch=i386 ?compare@?$collate@_W@std@@QBEHPB_W000@Z(ptr ptr ptr ptr ptr) collate_wchar_compare
 @ cdecl -arch=win64 ?compare@?$collate@_W@std@@QEBAHPEB_W000@Z(ptr ptr ptr ptr ptr) collate_wchar_compare
-@ stub -arch=win32 ?conj@std@@YA?AV?$complex@M@1@ABV21@@Z
-@ stub -arch=win64 ?conj@std@@YA?AV?$complex@M@1@AEBV21@@Z
-@ stub -arch=win32 ?conj@std@@YA?AV?$complex@N@1@ABV21@@Z
-@ stub -arch=win64 ?conj@std@@YA?AV?$complex@N@1@AEBV21@@Z
-@ stub -arch=win32 ?conj@std@@YA?AV?$complex@O@1@ABV21@@Z
-@ stub -arch=win64 ?conj@std@@YA?AV?$complex@O@1@AEBV21@@Z
+@ cdecl -arch=win32 ?conj@std@@YA?AV?$complex@M@1@ABV21@@Z(ptr ptr) complex_float_conj
+@ cdecl -arch=win64 ?conj@std@@YA?AV?$complex@M@1@AEBV21@@Z(ptr ptr) complex_float_conj
+@ cdecl -arch=win32 ?conj@std@@YA?AV?$complex@N@1@ABV21@@Z(ptr ptr) complex_double_conj
+@ cdecl -arch=win64 ?conj@std@@YA?AV?$complex@N@1@AEBV21@@Z(ptr ptr) complex_double_conj
+@ cdecl -arch=win32 ?conj@std@@YA?AV?$complex@O@1@ABV21@@Z(ptr ptr) complex_double_conj
+@ cdecl -arch=win64 ?conj@std@@YA?AV?$complex@O@1@AEBV21@@Z(ptr ptr) complex_double_conj
 @ cdecl -arch=arm ?copy@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@QBAIPADII@Z(ptr ptr long long) basic_string_char_copy
 @ thiscall -arch=i386 ?copy@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@QBEIPADII@Z(ptr ptr long long) basic_string_char_copy
 @ cdecl -arch=win64 ?copy@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@QEBA_KPEAD_K1@Z(ptr ptr long long) basic_string_char_copy
@@ -3427,18 +3427,18 @@
 @ cdecl ?cos@?$_Ctraits@M@std@@SAMM@Z(float) std_Ctraits_float_cos
 @ cdecl ?cos@?$_Ctraits@N@std@@SANN@Z(double) std_Ctraits_double_cos
 @ cdecl ?cos@?$_Ctraits@O@std@@SAOO@Z(double) std_Ctraits_long_double_cos
-@ stub -arch=win32 ?cos@std@@YA?AV?$complex@M@1@ABV21@@Z
-@ stub -arch=win64 ?cos@std@@YA?AV?$complex@M@1@AEBV21@@Z
-@ stub -arch=win32 ?cos@std@@YA?AV?$complex@N@1@ABV21@@Z
-@ stub -arch=win64 ?cos@std@@YA?AV?$complex@N@1@AEBV21@@Z
-@ stub -arch=win32 ?cos@std@@YA?AV?$complex@O@1@ABV21@@Z
-@ stub -arch=win64 ?cos@std@@YA?AV?$complex@O@1@AEBV21@@Z
-@ stub -arch=win32 ?cosh@std@@YA?AV?$complex@M@1@ABV21@@Z
-@ stub -arch=win64 ?cosh@std@@YA?AV?$complex@M@1@AEBV21@@Z
-@ stub -arch=win32 ?cosh@std@@YA?AV?$complex@N@1@ABV21@@Z
-@ stub -arch=win64 ?cosh@std@@YA?AV?$complex@N@1@AEBV21@@Z
-@ stub -arch=win32 ?cosh@std@@YA?AV?$complex@O@1@ABV21@@Z
-@ stub -arch=win64 ?cosh@std@@YA?AV?$complex@O@1@AEBV21@@Z
+@ cdecl -arch=win32 ?cos@std@@YA?AV?$complex@M@1@ABV21@@Z(ptr ptr) complex_float_cos
+@ cdecl -arch=win64 ?cos@std@@YA?AV?$complex@M@1@AEBV21@@Z(ptr ptr) complex_float_cos
+@ cdecl -arch=win32 ?cos@std@@YA?AV?$complex@N@1@ABV21@@Z(ptr ptr) complex_double_cos
+@ cdecl -arch=win64 ?cos@std@@YA?AV?$complex@N@1@AEBV21@@Z(ptr ptr) complex_double_cos
+@ cdecl -arch=win32 ?cos@std@@YA?AV?$complex@O@1@ABV21@@Z(ptr ptr) complex_double_cos
+@ cdecl -arch=win64 ?cos@std@@YA?AV?$complex@O@1@AEBV21@@Z(ptr ptr) complex_double_cos
+@ cdecl -arch=win32 ?cosh@std@@YA?AV?$complex@M@1@ABV21@@Z(ptr ptr) complex_float_cosh
+@ cdecl -arch=win64 ?cosh@std@@YA?AV?$complex@M@1@AEBV21@@Z(ptr ptr) complex_float_cosh
+@ cdecl -arch=win32 ?cosh@std@@YA?AV?$complex@N@1@ABV21@@Z(ptr ptr) complex_double_cosh
+@ cdecl -arch=win64 ?cosh@std@@YA?AV?$complex@N@1@AEBV21@@Z(ptr ptr) complex_double_cosh
+@ cdecl -arch=win32 ?cosh@std@@YA?AV?$complex@O@1@ABV21@@Z(ptr ptr) complex_double_cosh
+@ cdecl -arch=win64 ?cosh@std@@YA?AV?$complex@O@1@AEBV21@@Z(ptr ptr) complex_double_cosh
 @ extern ?cout@std@@3V?$basic_ostream@DU?$char_traits@D@std@@@1@A cout
 @ cdecl -arch=arm ?curr_symbol@?$_Mpunct@D@std@@QBA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@2@XZ(ptr ptr) mpunct_char_curr_symbol
 @ thiscall -arch=i386 ?curr_symbol@?$_Mpunct@D@std@@QBE?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@2@XZ(ptr ptr) mpunct_char_curr_symbol
@@ -4275,12 +4275,12 @@
 @ cdecl ?exp@?$_Ctraits@M@std@@SAMM@Z(float) std_Ctraits_float_exp
 @ cdecl ?exp@?$_Ctraits@N@std@@SANN@Z(double) std_Ctraits_double_exp
 @ cdecl ?exp@?$_Ctraits@O@std@@SAOO@Z(double) std_Ctraits_long_double_exp
-@ stub -arch=win32 ?exp@std@@YA?AV?$complex@M@1@ABV21@@Z
-@ stub -arch=win64 ?exp@std@@YA?AV?$complex@M@1@AEBV21@@Z
-@ stub -arch=win32 ?exp@std@@YA?AV?$complex@N@1@ABV21@@Z
-@ stub -arch=win64 ?exp@std@@YA?AV?$complex@N@1@AEBV21@@Z
-@ stub -arch=win32 ?exp@std@@YA?AV?$complex@O@1@ABV21@@Z
-@ stub -arch=win64 ?exp@std@@YA?AV?$complex@O@1@AEBV21@@Z
+@ cdecl -arch=win32 ?exp@std@@YA?AV?$complex@M@1@ABV21@@Z(ptr ptr) complex_float_exp
+@ cdecl -arch=win64 ?exp@std@@YA?AV?$complex@M@1@AEBV21@@Z(ptr ptr) complex_float_exp
+@ cdecl -arch=win32 ?exp@std@@YA?AV?$complex@N@1@ABV21@@Z(ptr ptr) complex_double_exp
+@ cdecl -arch=win64 ?exp@std@@YA?AV?$complex@N@1@AEBV21@@Z(ptr ptr) complex_double_exp
+@ cdecl -arch=win32 ?exp@std@@YA?AV?$complex@O@1@ABV21@@Z(ptr ptr) complex_double_exp
+@ cdecl -arch=win64 ?exp@std@@YA?AV?$complex@O@1@AEBV21@@Z(ptr ptr) complex_double_exp
 @ cdecl -arch=arm ?fail@ios_base@std@@QBA_NXZ(ptr) ios_base_fail
 @ thiscall -arch=i386 ?fail@ios_base@std@@QBE_NXZ(ptr) ios_base_fail
 @ cdecl -arch=win64 ?fail@ios_base@std@@QEBA_NXZ(ptr) ios_base_fail
@@ -4921,29 +4921,29 @@
 @ cdecl -arch=arm ?ignore@?$basic_istream@_WU?$char_traits@_W@std@@@std@@QAAAAV12@HG@Z(ptr long long) basic_istream_wchar_ignore
 @ thiscall -arch=i386 ?ignore@?$basic_istream@_WU?$char_traits@_W@std@@@std@@QAEAAV12@HG@Z(ptr long long) basic_istream_wchar_ignore
 @ stub -arch=arm ?imag@?$_Complex_base@M@std@@QAAMABM@Z
-@ stub -arch=i386 ?imag@?$_Complex_base@M@std@@QAEMABM@Z
-@ stub -arch=win64 ?imag@?$_Complex_base@M@std@@QEAAMAEBM@Z
+@ thiscall -arch=i386 ?imag@?$_Complex_base@M@std@@QAEMABM@Z(ptr ptr) complex_float_imag_set
+@ cdecl -arch=win64 ?imag@?$_Complex_base@M@std@@QEAAMAEBM@Z(ptr ptr) complex_float_imag_set
 @ stub -arch=arm ?imag@?$_Complex_base@M@std@@QBAMXZ
-@ stub -arch=i386 ?imag@?$_Complex_base@M@std@@QBEMXZ
-@ stub -arch=win64 ?imag@?$_Complex_base@M@std@@QEBAMXZ
+@ thiscall -arch=i386 ?imag@?$_Complex_base@M@std@@QBEMXZ(ptr) complex_float_imag_get
+@ cdecl -arch=win64 ?imag@?$_Complex_base@M@std@@QEBAMXZ(ptr) complex_float_imag_get
 @ stub -arch=arm ?imag@?$_Complex_base@N@std@@QAANABN@Z
-@ stub -arch=i386 ?imag@?$_Complex_base@N@std@@QAENABN@Z
-@ stub -arch=win64 ?imag@?$_Complex_base@N@std@@QEAANAEBN@Z
+@ thiscall -arch=i386 ?imag@?$_Complex_base@N@std@@QAENABN@Z(ptr ptr) complex_double_imag_set
+@ cdecl -arch=win64 ?imag@?$_Complex_base@N@std@@QEAANAEBN@Z(ptr ptr) complex_double_imag_set
 @ stub -arch=arm ?imag@?$_Complex_base@N@std@@QBANXZ
-@ stub -arch=i386 ?imag@?$_Complex_base@N@std@@QBENXZ
-@ stub -arch=win64 ?imag@?$_Complex_base@N@std@@QEBANXZ
+@ thiscall -arch=i386 ?imag@?$_Complex_base@N@std@@QBENXZ(ptr) complex_double_imag_get
+@ cdecl -arch=win64 ?imag@?$_Complex_base@N@std@@QEBANXZ(ptr) complex_double_imag_get
 @ stub -arch=arm ?imag@?$_Complex_base@O@std@@QAAOABO@Z
-@ stub -arch=i386 ?imag@?$_Complex_base@O@std@@QAEOABO@Z
-@ stub -arch=win64 ?imag@?$_Complex_base@O@std@@QEAAOAEBO@Z
+@ thiscall -arch=i386 ?imag@?$_Complex_base@O@std@@QAEOABO@Z(ptr ptr) complex_double_imag_set
+@ cdecl -arch=win64 ?imag@?$_Complex_base@O@std@@QEAAOAEBO@Z(ptr ptr) complex_double_imag_set
 @ stub -arch=arm ?imag@?$_Complex_base@O@std@@QBAOXZ
-@ stub -arch=i386 ?imag@?$_Complex_base@O@std@@QBEOXZ
-@ stub -arch=win64 ?imag@?$_Complex_base@O@std@@QEBAOXZ
-@ stub -arch=win32 ?imag@std@@YAMABV?$complex@M@1@@Z
-@ stub -arch=win64 ?imag@std@@YAMAEBV?$complex@M@1@@Z
-@ stub -arch=win32 ?imag@std@@YANABV?$complex@N@1@@Z
-@ stub -arch=win64 ?imag@std@@YANAEBV?$complex@N@1@@Z
-@ stub -arch=win32 ?imag@std@@YAOABV?$complex@O@1@@Z
-@ stub -arch=win64 ?imag@std@@YAOAEBV?$complex@O@1@@Z
+@ thiscall -arch=i386 ?imag@?$_Complex_base@O@std@@QBEOXZ(ptr) complex_double_imag_get
+@ cdecl -arch=win64 ?imag@?$_Complex_base@O@std@@QEBAOXZ(ptr) complex_double_imag_get
+@ cdecl -arch=win32 ?imag@std@@YAMABV?$complex@M@1@@Z(ptr) complex_float_imag
+@ cdecl -arch=win64 ?imag@std@@YAMAEBV?$complex@M@1@@Z(ptr) complex_float_imag
+@ cdecl -arch=win32 ?imag@std@@YANABV?$complex@N@1@@Z(ptr) complex_double_imag
+@ cdecl -arch=win64 ?imag@std@@YANAEBV?$complex@N@1@@Z(ptr) complex_double_imag
+@ cdecl -arch=win32 ?imag@std@@YAOABV?$complex@O@1@@Z(ptr) complex_double_imag
+@ cdecl -arch=win64 ?imag@std@@YAOAEBV?$complex@O@1@@Z(ptr) complex_double_imag
 @ cdecl -arch=arm ?imbue@?$basic_filebuf@DU?$char_traits@D@std@@@std@@MAAXABVlocale@2@@Z(ptr ptr) basic_filebuf_char_imbue
 @ thiscall -arch=i386 ?imbue@?$basic_filebuf@DU?$char_traits@D@std@@@std@@MAEXABVlocale@2@@Z(ptr ptr) basic_filebuf_char_imbue
 @ cdecl -arch=win64 ?imbue@?$basic_filebuf@DU?$char_traits@D@std@@@std@@MEAAXAEBVlocale@2@@Z(ptr ptr) basic_filebuf_char_imbue
@@ -5232,21 +5232,21 @@
 @ cdecl -arch=arm ?length@?$codecvt@_WDH@std@@QBAHABHPBD1I@Z(ptr ptr str str long) codecvt_wchar_length
 @ thiscall -arch=i386 ?length@?$codecvt@_WDH@std@@QBEHABHPBD1I@Z(ptr ptr str str long) codecvt_wchar_length
 @ cdecl -arch=win64 ?length@?$codecvt@_WDH@std@@QEBAHAEBHPEBD1_K@Z(ptr ptr str str long) codecvt_wchar_length
-@ stub -arch=win32 ?log10@std@@YA?AV?$complex@M@1@ABV21@@Z
-@ stub -arch=win64 ?log10@std@@YA?AV?$complex@M@1@AEBV21@@Z
-@ stub -arch=win32 ?log10@std@@YA?AV?$complex@N@1@ABV21@@Z
-@ stub -arch=win64 ?log10@std@@YA?AV?$complex@N@1@AEBV21@@Z
-@ stub -arch=win32 ?log10@std@@YA?AV?$complex@O@1@ABV21@@Z
-@ stub -arch=win64 ?log10@std@@YA?AV?$complex@O@1@AEBV21@@Z
+@ cdecl -arch=win32 ?log10@std@@YA?AV?$complex@M@1@ABV21@@Z(ptr ptr) complex_float_log10
+@ cdecl -arch=win64 ?log10@std@@YA?AV?$complex@M@1@AEBV21@@Z(ptr ptr) complex_float_log10
+@ cdecl -arch=win32 ?log10@std@@YA?AV?$complex@N@1@ABV21@@Z(ptr ptr) complex_double_log10
+@ cdecl -arch=win64 ?log10@std@@YA?AV?$complex@N@1@AEBV21@@Z(ptr ptr) complex_double_log10
+@ cdecl -arch=win32 ?log10@std@@YA?AV?$complex@O@1@ABV21@@Z(ptr ptr) complex_double_log10
+@ cdecl -arch=win64 ?log10@std@@YA?AV?$complex@O@1@AEBV21@@Z(ptr ptr) complex_double_log10
 @ cdecl ?log@?$_Ctraits@M@std@@SAMM@Z(float) std_Ctraits_float_log
 @ cdecl ?log@?$_Ctraits@N@std@@SANN@Z(double) std_Ctraits_double_log
 @ cdecl ?log@?$_Ctraits@O@std@@SAOO@Z(double) std_Ctraits_long_double_log
-@ stub -arch=win32 ?log@std@@YA?AV?$complex@M@1@ABV21@@Z
-@ stub -arch=win64 ?log@std@@YA?AV?$complex@M@1@AEBV21@@Z
-@ stub -arch=win32 ?log@std@@YA?AV?$complex@N@1@ABV21@@Z
-@ stub -arch=win64 ?log@std@@YA?AV?$complex@N@1@AEBV21@@Z
-@ stub -arch=win32 ?log@std@@YA?AV?$complex@O@1@ABV21@@Z
-@ stub -arch=win64 ?log@std@@YA?AV?$complex@O@1@AEBV21@@Z
+@ cdecl -arch=win32 ?log@std@@YA?AV?$complex@M@1@ABV21@@Z(ptr ptr) complex_float_log
+@ cdecl -arch=win64 ?log@std@@YA?AV?$complex@M@1@AEBV21@@Z(ptr ptr) complex_float_log
+@ cdecl -arch=win32 ?log@std@@YA?AV?$complex@N@1@ABV21@@Z(ptr ptr) complex_double_log
+@ cdecl -arch=win64 ?log@std@@YA?AV?$complex@N@1@AEBV21@@Z(ptr ptr) complex_double_log
+@ cdecl -arch=win32 ?log@std@@YA?AV?$complex@O@1@ABV21@@Z(ptr ptr) complex_double_log
+@ cdecl -arch=win64 ?log@std@@YA?AV?$complex@O@1@AEBV21@@Z(ptr ptr) complex_double_log
 @ cdecl -arch=win32 ?lt@?$char_traits@D@std@@SA_NABD0@Z(ptr ptr) MSVCP_char_traits_lt
 @ cdecl -arch=win64 ?lt@?$char_traits@D@std@@SA_NAEBD0@Z(ptr ptr) MSVCP_char_traits_lt
 @ cdecl -arch=win32 ?lt@?$char_traits@G@std@@SA_NABG0@Z(ptr ptr) MSVCP_char_traits_short_lt
@@ -5367,12 +5367,12 @@
 @ cdecl -arch=arm ?negative_sign@?$_Mpunct@_W@std@@QBA?AV?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@2@XZ(ptr ptr) mpunct_wchar_negative_sign
 @ thiscall -arch=i386 ?negative_sign@?$_Mpunct@_W@std@@QBE?AV?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@2@XZ(ptr ptr) mpunct_wchar_negative_sign
 @ cdecl -arch=win64 ?negative_sign@?$_Mpunct@_W@std@@QEBA?AV?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@2@XZ(ptr ptr) mpunct_wchar_negative_sign
-@ stub -arch=win32 ?norm@std@@YAMABV?$complex@M@1@@Z
-@ stub -arch=win64 ?norm@std@@YAMAEBV?$complex@M@1@@Z
-@ stub -arch=win32 ?norm@std@@YANABV?$complex@N@1@@Z
-@ stub -arch=win64 ?norm@std@@YANAEBV?$complex@N@1@@Z
-@ stub -arch=win32 ?norm@std@@YAOABV?$complex@O@1@@Z
-@ stub -arch=win64 ?norm@std@@YAOAEBV?$complex@O@1@@Z
+@ cdecl -arch=win32 ?norm@std@@YAMABV?$complex@M@1@@Z(ptr) complex_float_norm
+@ cdecl -arch=win64 ?norm@std@@YAMAEBV?$complex@M@1@@Z(ptr) complex_float_norm
+@ cdecl -arch=win32 ?norm@std@@YANABV?$complex@N@1@@Z(ptr) complex_double_norm
+@ cdecl -arch=win64 ?norm@std@@YANAEBV?$complex@N@1@@Z(ptr) complex_double_norm
+@ cdecl -arch=win32 ?norm@std@@YAOABV?$complex@O@1@@Z(ptr) complex_double_norm
+@ cdecl -arch=win64 ?norm@std@@YAOAEBV?$complex@O@1@@Z(ptr) complex_double_norm
 @ cdecl -arch=win32 ?not_eof@?$char_traits@D@std@@SAHABH@Z(ptr) MSVCP_char_traits_char_not_eof
 @ cdecl -arch=win64 ?not_eof@?$char_traits@D@std@@SAHAEBH@Z(ptr) MSVCP_char_traits_char_not_eof
 @ cdecl -arch=win32 ?not_eof@?$char_traits@G@std@@SAGABG@Z(ptr) MSVCP_char_traits_short_not_eof
@@ -5576,18 +5576,18 @@
 @ cdecl -arch=arm ?peek@?$basic_istream@_WU?$char_traits@_W@std@@@std@@QAAGXZ(ptr) basic_istream_wchar_peek
 @ thiscall -arch=i386 ?peek@?$basic_istream@_WU?$char_traits@_W@std@@@std@@QAEGXZ(ptr) basic_istream_wchar_peek
 @ cdecl -arch=win64 ?peek@?$basic_istream@_WU?$char_traits@_W@std@@@std@@QEAAGXZ(ptr) basic_istream_wchar_peek
-@ stub -arch=win32 ?polar@std@@YA?AV?$complex@M@1@ABM0@Z
-@ stub -arch=win64 ?polar@std@@YA?AV?$complex@M@1@AEBM0@Z
-@ stub -arch=win32 ?polar@std@@YA?AV?$complex@M@1@ABM@Z
-@ stub -arch=win64 ?polar@std@@YA?AV?$complex@M@1@AEBM@Z
-@ stub -arch=win32 ?polar@std@@YA?AV?$complex@N@1@ABN0@Z
-@ stub -arch=win64 ?polar@std@@YA?AV?$complex@N@1@AEBN0@Z
-@ stub -arch=win32 ?polar@std@@YA?AV?$complex@N@1@ABN@Z
-@ stub -arch=win64 ?polar@std@@YA?AV?$complex@N@1@AEBN@Z
-@ stub -arch=win32 ?polar@std@@YA?AV?$complex@O@1@ABO0@Z
-@ stub -arch=win64 ?polar@std@@YA?AV?$complex@O@1@AEBO0@Z
-@ stub -arch=win32 ?polar@std@@YA?AV?$complex@O@1@ABO@Z
-@ stub -arch=win64 ?polar@std@@YA?AV?$complex@O@1@AEBO@Z
+@ cdecl -arch=win32 ?polar@std@@YA?AV?$complex@M@1@ABM0@Z(ptr ptr ptr) complex_float_polar_theta
+@ cdecl -arch=win64 ?polar@std@@YA?AV?$complex@M@1@AEBM0@Z(ptr ptr ptr) complex_float_polar_theta
+@ cdecl -arch=win32 ?polar@std@@YA?AV?$complex@M@1@ABM@Z(ptr ptr) complex_float_polar
+@ cdecl -arch=win64 ?polar@std@@YA?AV?$complex@M@1@AEBM@Z(ptr ptr) complex_float_polar
+@ cdecl -arch=win32 ?polar@std@@YA?AV?$complex@N@1@ABN0@Z(ptr ptr ptr) complex_double_polar_theta
+@ cdecl -arch=win64 ?polar@std@@YA?AV?$complex@N@1@AEBN0@Z(ptr ptr ptr) complex_double_polar_theta
+@ cdecl -arch=win32 ?polar@std@@YA?AV?$complex@N@1@ABN@Z(ptr ptr) complex_double_polar
+@ cdecl -arch=win64 ?polar@std@@YA?AV?$complex@N@1@AEBN@Z(ptr ptr) complex_double_polar
+@ cdecl -arch=win32 ?polar@std@@YA?AV?$complex@O@1@ABO0@Z(ptr ptr ptr) complex_double_polar_theta
+@ cdecl -arch=win64 ?polar@std@@YA?AV?$complex@O@1@AEBO0@Z(ptr ptr ptr) complex_double_polar_theta
+@ cdecl -arch=win32 ?polar@std@@YA?AV?$complex@O@1@ABO@Z(ptr ptr) complex_double_polar
+@ cdecl -arch=win64 ?polar@std@@YA?AV?$complex@O@1@AEBO@Z(ptr ptr) complex_double_polar
 @ cdecl -arch=arm ?pos_format@?$_Mpunct@D@std@@QBA?AUpattern@money_base@2@XZ(ptr ptr) mpunct_char_pos_format
 @ thiscall -arch=i386 ?pos_format@?$_Mpunct@D@std@@QBE?AUpattern@money_base@2@XZ(ptr ptr) mpunct_char_pos_format
 @ cdecl -arch=win64 ?pos_format@?$_Mpunct@D@std@@QEBA?AUpattern@money_base@2@XZ(ptr ptr) mpunct_char_pos_format
@@ -5609,28 +5609,28 @@
 @ cdecl ?pow@?$_Ctraits@M@std@@SAMMM@Z(float float) std_Ctraits_float_pow
 @ cdecl ?pow@?$_Ctraits@N@std@@SANNN@Z(double double) std_Ctraits_double_pow
 @ cdecl ?pow@?$_Ctraits@O@std@@SAOOO@Z(double double) std_Ctraits_long_double_pow
-@ stub -arch=win32 ?pow@std@@YA?AV?$complex@M@1@ABMABV21@@Z
-@ stub -arch=win64 ?pow@std@@YA?AV?$complex@M@1@AEBMAEBV21@@Z
-@ stub -arch=win32 ?pow@std@@YA?AV?$complex@M@1@ABV21@0@Z
-@ stub -arch=win64 ?pow@std@@YA?AV?$complex@M@1@AEBV21@0@Z
-@ stub -arch=win32 ?pow@std@@YA?AV?$complex@M@1@ABV21@ABM@Z
-@ stub -arch=win64 ?pow@std@@YA?AV?$complex@M@1@AEBV21@AEBM@Z
+@ cdecl -arch=win32 ?pow@std@@YA?AV?$complex@M@1@ABMABV21@@Z(ptr ptr ptr) complex_float_pow_fc
+@ cdecl -arch=win64 ?pow@std@@YA?AV?$complex@M@1@AEBMAEBV21@@Z(ptr ptr ptr) complex_float_pow_fc
+@ cdecl -arch=win32 ?pow@std@@YA?AV?$complex@M@1@ABV21@0@Z(ptr ptr ptr) complex_float_pow
+@ cdecl -arch=win64 ?pow@std@@YA?AV?$complex@M@1@AEBV21@0@Z(ptr ptr ptr) complex_float_pow
+@ cdecl -arch=win32 ?pow@std@@YA?AV?$complex@M@1@ABV21@ABM@Z(ptr ptr ptr) complex_float_pow_cf
+@ cdecl -arch=win64 ?pow@std@@YA?AV?$complex@M@1@AEBV21@AEBM@Z(ptr ptr ptr) complex_float_pow_cf
 @ cdecl -arch=win32 ?pow@std@@YA?AV?$complex@M@1@ABV21@H@Z(ptr ptr long) complex_float_pow_ci
 @ cdecl -arch=win64 ?pow@std@@YA?AV?$complex@M@1@AEBV21@H@Z(ptr ptr long) complex_float_pow_ci
-@ stub -arch=win32 ?pow@std@@YA?AV?$complex@N@1@ABNABV21@@Z
-@ stub -arch=win64 ?pow@std@@YA?AV?$complex@N@1@AEBNAEBV21@@Z
-@ stub -arch=win32 ?pow@std@@YA?AV?$complex@N@1@ABV21@0@Z
-@ stub -arch=win64 ?pow@std@@YA?AV?$complex@N@1@AEBV21@0@Z
-@ stub -arch=win32 ?pow@std@@YA?AV?$complex@N@1@ABV21@ABN@Z
+@ cdecl -arch=win32 ?pow@std@@YA?AV?$complex@N@1@ABNABV21@@Z(ptr ptr ptr) complex_double_pow_dc
+@ cdecl -arch=win64 ?pow@std@@YA?AV?$complex@N@1@AEBNAEBV21@@Z(ptr ptr ptr) complex_double_pow_dc
+@ cdecl -arch=win32 ?pow@std@@YA?AV?$complex@N@1@ABV21@0@Z(ptr ptr ptr) complex_double_pow
+@ cdecl -arch=win64 ?pow@std@@YA?AV?$complex@N@1@AEBV21@0@Z(ptr ptr ptr) complex_double_pow
+@ cdecl -arch=win32 ?pow@std@@YA?AV?$complex@N@1@ABV21@ABN@Z(ptr ptr ptr) complex_double_pow_cd
 @ cdecl -arch=win64 ?pow@std@@YA?AV?$complex@N@1@AEBV21@AEBN@Z(ptr ptr long) complex_double_pow_ci
 @ cdecl -arch=win32 ?pow@std@@YA?AV?$complex@N@1@ABV21@H@Z(ptr ptr long) complex_double_pow_ci
-@ stub -arch=win64 ?pow@std@@YA?AV?$complex@N@1@AEBV21@H@Z
-@ stub -arch=win32 ?pow@std@@YA?AV?$complex@O@1@ABOABV21@@Z
-@ stub -arch=win64 ?pow@std@@YA?AV?$complex@O@1@AEBOAEBV21@@Z
-@ stub -arch=win32 ?pow@std@@YA?AV?$complex@O@1@ABV21@0@Z
-@ stub -arch=win64 ?pow@std@@YA?AV?$complex@O@1@AEBV21@0@Z
-@ stub -arch=win32 ?pow@std@@YA?AV?$complex@O@1@ABV21@ABO@Z
-@ stub -arch=win64 ?pow@std@@YA?AV?$complex@O@1@AEBV21@AEBO@Z
+@ cdecl -arch=win64 ?pow@std@@YA?AV?$complex@N@1@AEBV21@H@Z(ptr ptr long) complex_double_pow_ci
+@ cdecl -arch=win32 ?pow@std@@YA?AV?$complex@O@1@ABOABV21@@Z(ptr ptr ptr) complex_double_pow_dc
+@ cdecl -arch=win64 ?pow@std@@YA?AV?$complex@O@1@AEBOAEBV21@@Z(ptr ptr ptr) complex_double_pow_dc
+@ cdecl -arch=win32 ?pow@std@@YA?AV?$complex@O@1@ABV21@0@Z(ptr ptr ptr) complex_double_pow
+@ cdecl -arch=win64 ?pow@std@@YA?AV?$complex@O@1@AEBV21@0@Z(ptr ptr ptr) complex_double_pow
+@ cdecl -arch=win32 ?pow@std@@YA?AV?$complex@O@1@ABV21@ABO@Z(ptr ptr ptr) complex_double_pow_cd
+@ cdecl -arch=win64 ?pow@std@@YA?AV?$complex@O@1@AEBV21@AEBO@Z(ptr ptr ptr) complex_double_pow_cd
 @ cdecl -arch=win32 ?pow@std@@YA?AV?$complex@O@1@ABV21@H@Z(ptr ptr long) complex_double_pow_ci
 @ cdecl -arch=win64 ?pow@std@@YA?AV?$complex@O@1@AEBV21@H@Z(ptr ptr long) complex_double_pow_ci
 @ cdecl -arch=arm ?pptr@?$basic_streambuf@DU?$char_traits@D@std@@@std@@IBAPADXZ(ptr) basic_streambuf_char_pptr
@@ -5974,29 +5974,29 @@
 @ thiscall -arch=i386 ?readsome@?$basic_istream@_WU?$char_traits@_W@std@@@std@@QAEHPA_WH@Z(ptr ptr long) basic_istream_wchar_readsome
 @ cdecl -arch=win64 ?readsome@?$basic_istream@_WU?$char_traits@_W@std@@@std@@QEAA_JPEA_W_J@Z(ptr ptr long) basic_istream_wchar_readsome
 @ stub -arch=arm ?real@?$_Complex_base@M@std@@QAAMABM@Z
-@ stub -arch=i386 ?real@?$_Complex_base@M@std@@QAEMABM@Z
-@ stub -arch=win64 ?real@?$_Complex_base@M@std@@QEAAMAEBM@Z
+@ thiscall -arch=i386 ?real@?$_Complex_base@M@std@@QAEMABM@Z(ptr ptr) complex_float_real_set
+@ cdecl -arch=win64 ?real@?$_Complex_base@M@std@@QEAAMAEBM@Z(ptr ptr) complex_float_real_set
 @ stub -arch=arm ?real@?$_Complex_base@M@std@@QBAMXZ
-@ stub -arch=i386 ?real@?$_Complex_base@M@std@@QBEMXZ
-@ stub -arch=win64 ?real@?$_Complex_base@M@std@@QEBAMXZ
+@ thiscall -arch=i386 ?real@?$_Complex_base@M@std@@QBEMXZ(ptr) complex_float_real_get
+@ cdecl -arch=win64 ?real@?$_Complex_base@M@std@@QEBAMXZ(ptr) complex_float_real_get
 @ stub -arch=arm ?real@?$_Complex_base@N@std@@QAANABN@Z
-@ stub -arch=i386 ?real@?$_Complex_base@N@std@@QAENABN@Z
-@ stub -arch=win64 ?real@?$_Complex_base@N@std@@QEAANAEBN@Z
+@ thiscall -arch=i386 ?real@?$_Complex_base@N@std@@QAENABN@Z(ptr ptr) complex_double_real_set
+@ cdecl -arch=win64 ?real@?$_Complex_base@N@std@@QEAANAEBN@Z(ptr ptr) complex_double_real_set
 @ stub -arch=arm ?real@?$_Complex_base@N@std@@QBANXZ
-@ stub -arch=i386 ?real@?$_Complex_base@N@std@@QBENXZ
-@ stub -arch=win64 ?real@?$_Complex_base@N@std@@QEBANXZ
+@ thiscall -arch=i386 ?real@?$_Complex_base@N@std@@QBENXZ(ptr) complex_double_real_get
+@ cdecl -arch=win64 ?real@?$_Complex_base@N@std@@QEBANXZ(ptr) complex_double_real_get
 @ stub -arch=arm ?real@?$_Complex_base@O@std@@QAAOABO@Z
-@ stub -arch=i386 ?real@?$_Complex_base@O@std@@QAEOABO@Z
-@ stub -arch=win64 ?real@?$_Complex_base@O@std@@QEAAOAEBO@Z
+@ thiscall -arch=i386 ?real@?$_Complex_base@O@std@@QAEOABO@Z(ptr ptr) complex_double_real_set
+@ cdecl -arch=win64 ?real@?$_Complex_base@O@std@@QEAAOAEBO@Z(ptr ptr) complex_double_real_set
 @ stub -arch=arm ?real@?$_Complex_base@O@std@@QBAOXZ
-@ stub -arch=i386 ?real@?$_Complex_base@O@std@@QBEOXZ
-@ stub -arch=win64 ?real@?$_Complex_base@O@std@@QEBAOXZ
-@ stub -arch=win32 ?real@std@@YAMABV?$complex@M@1@@Z
-@ stub -arch=win64 ?real@std@@YAMAEBV?$complex@M@1@@Z
-@ stub -arch=win32 ?real@std@@YANABV?$complex@N@1@@Z
-@ stub -arch=win64 ?real@std@@YANAEBV?$complex@N@1@@Z
-@ stub -arch=win32 ?real@std@@YAOABV?$complex@O@1@@Z
-@ stub -arch=win64 ?real@std@@YAOAEBV?$complex@O@1@@Z
+@ thiscall -arch=i386 ?real@?$_Complex_base@O@std@@QBEOXZ(ptr) complex_double_real_get
+@ cdecl -arch=win64 ?real@?$_Complex_base@O@std@@QEBAOXZ(ptr) complex_double_real_get
+@ cdecl -arch=win32 ?real@std@@YAMABV?$complex@M@1@@Z(ptr) complex_float_real
+@ cdecl -arch=win64 ?real@std@@YAMAEBV?$complex@M@1@@Z(ptr) complex_float_real
+@ cdecl -arch=win32 ?real@std@@YANABV?$complex@N@1@@Z(ptr) complex_double_real
+@ cdecl -arch=win64 ?real@std@@YANAEBV?$complex@N@1@@Z(ptr) complex_double_real
+@ cdecl -arch=win32 ?real@std@@YAOABV?$complex@O@1@@Z(ptr) complex_double_real
+@ cdecl -arch=win64 ?real@std@@YAOAEBV?$complex@O@1@@Z(ptr) complex_double_real
 @ cdecl -arch=arm ?register_callback@ios_base@std@@QAAXP6AXW4event@12@AAV12@H@ZH@Z(ptr ptr long) ios_base_register_callback
 @ thiscall -arch=i386 ?register_callback@ios_base@std@@QAEXP6AXW4event@12@AAV12@H@ZH@Z(ptr ptr long) ios_base_register_callback
 @ cdecl -arch=win64 ?register_callback@ios_base@std@@QEAAXP6AXW4event@12@AEAV12@H@ZH@Z(ptr ptr long) ios_base_register_callback
@@ -6430,18 +6430,18 @@
 @ cdecl ?sin@?$_Ctraits@M@std@@SAMM@Z(float) std_Ctraits_float_sin
 @ cdecl ?sin@?$_Ctraits@N@std@@SANN@Z(double) std_Ctraits_double_sin
 @ cdecl ?sin@?$_Ctraits@O@std@@SAOO@Z(double) std_Ctraits_long_double_sin
-@ stub -arch=win32 ?sin@std@@YA?AV?$complex@M@1@ABV21@@Z
-@ stub -arch=win64 ?sin@std@@YA?AV?$complex@M@1@AEBV21@@Z
-@ stub -arch=win32 ?sin@std@@YA?AV?$complex@N@1@ABV21@@Z
-@ stub -arch=win64 ?sin@std@@YA?AV?$complex@N@1@AEBV21@@Z
-@ stub -arch=win32 ?sin@std@@YA?AV?$complex@O@1@ABV21@@Z
-@ stub -arch=win64 ?sin@std@@YA?AV?$complex@O@1@AEBV21@@Z
-@ stub -arch=win32 ?sinh@std@@YA?AV?$complex@M@1@ABV21@@Z
-@ stub -arch=win64 ?sinh@std@@YA?AV?$complex@M@1@AEBV21@@Z
-@ stub -arch=win32 ?sinh@std@@YA?AV?$complex@N@1@ABV21@@Z
-@ stub -arch=win64 ?sinh@std@@YA?AV?$complex@N@1@AEBV21@@Z
-@ stub -arch=win32 ?sinh@std@@YA?AV?$complex@O@1@ABV21@@Z
-@ stub -arch=win64 ?sinh@std@@YA?AV?$complex@O@1@AEBV21@@Z
+@ cdecl -arch=win32 ?sin@std@@YA?AV?$complex@M@1@ABV21@@Z(ptr ptr) complex_float_sin
+@ cdecl -arch=win64 ?sin@std@@YA?AV?$complex@M@1@AEBV21@@Z(ptr ptr) complex_float_sin
+@ cdecl -arch=win32 ?sin@std@@YA?AV?$complex@N@1@ABV21@@Z(ptr ptr) complex_double_sin
+@ cdecl -arch=win64 ?sin@std@@YA?AV?$complex@N@1@AEBV21@@Z(ptr ptr) complex_double_sin
+@ cdecl -arch=win32 ?sin@std@@YA?AV?$complex@O@1@ABV21@@Z(ptr ptr) complex_double_sin
+@ cdecl -arch=win64 ?sin@std@@YA?AV?$complex@O@1@AEBV21@@Z(ptr ptr) complex_double_sin
+@ cdecl -arch=win32 ?sinh@std@@YA?AV?$complex@M@1@ABV21@@Z(ptr ptr) complex_float_sinh
+@ cdecl -arch=win64 ?sinh@std@@YA?AV?$complex@M@1@AEBV21@@Z(ptr ptr) complex_float_sinh
+@ cdecl -arch=win32 ?sinh@std@@YA?AV?$complex@N@1@ABV21@@Z(ptr ptr) complex_double_sinh
+@ cdecl -arch=win64 ?sinh@std@@YA?AV?$complex@N@1@AEBV21@@Z(ptr ptr) complex_double_sinh
+@ cdecl -arch=win32 ?sinh@std@@YA?AV?$complex@O@1@ABV21@@Z(ptr ptr) complex_double_sinh
+@ cdecl -arch=win64 ?sinh@std@@YA?AV?$complex@O@1@AEBV21@@Z(ptr ptr) complex_double_sinh
 @ cdecl -arch=arm ?size@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@QBAIXZ(ptr) MSVCP_basic_string_char_length
 @ thiscall -arch=i386 ?size@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@QBEIXZ(ptr) MSVCP_basic_string_char_length
 @ cdecl -arch=win64 ?size@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@QEBA_KXZ(ptr) MSVCP_basic_string_char_length
@@ -6490,12 +6490,12 @@
 @ cdecl ?sqrt@?$_Ctraits@M@std@@SAMM@Z(float) std_Ctraits_float_sqrt
 @ cdecl ?sqrt@?$_Ctraits@N@std@@SANN@Z(double) std_Ctraits_double_sqrt
 @ cdecl ?sqrt@?$_Ctraits@O@std@@SAOO@Z(double) std_Ctraits_long_double_sqrt
-@ stub -arch=win32 ?sqrt@std@@YA?AV?$complex@M@1@ABV21@@Z
-@ stub -arch=win64 ?sqrt@std@@YA?AV?$complex@M@1@AEBV21@@Z
-@ stub -arch=win32 ?sqrt@std@@YA?AV?$complex@N@1@ABV21@@Z
-@ stub -arch=win64 ?sqrt@std@@YA?AV?$complex@N@1@AEBV21@@Z
-@ stub -arch=win32 ?sqrt@std@@YA?AV?$complex@O@1@ABV21@@Z
-@ stub -arch=win64 ?sqrt@std@@YA?AV?$complex@O@1@AEBV21@@Z
+@ cdecl -arch=win32 ?sqrt@std@@YA?AV?$complex@M@1@ABV21@@Z(ptr ptr) complex_float_sqrt
+@ cdecl -arch=win64 ?sqrt@std@@YA?AV?$complex@M@1@AEBV21@@Z(ptr ptr) complex_float_sqrt
+@ cdecl -arch=win32 ?sqrt@std@@YA?AV?$complex@N@1@ABV21@@Z(ptr ptr) complex_double_sqrt
+@ cdecl -arch=win64 ?sqrt@std@@YA?AV?$complex@N@1@AEBV21@@Z(ptr ptr) complex_double_sqrt
+@ cdecl -arch=win32 ?sqrt@std@@YA?AV?$complex@O@1@ABV21@@Z(ptr ptr) complex_double_sqrt
+@ cdecl -arch=win64 ?sqrt@std@@YA?AV?$complex@O@1@AEBV21@@Z(ptr ptr) complex_double_sqrt
 @ cdecl -arch=arm ?stossc@?$basic_streambuf@DU?$char_traits@D@std@@@std@@QAAXXZ(ptr) basic_streambuf_char_stossc
 @ thiscall -arch=i386 ?stossc@?$basic_streambuf@DU?$char_traits@D@std@@@std@@QAEXXZ(ptr) basic_streambuf_char_stossc
 @ cdecl -arch=win64 ?stossc@?$basic_streambuf@DU?$char_traits@D@std@@@std@@QEAAXXZ(ptr) basic_streambuf_char_stossc
@@ -6640,12 +6640,12 @@
 @ cdecl ?tan@?$_Ctraits@M@std@@SAMM@Z(float) std_Ctraits_float_tan
 @ cdecl ?tan@?$_Ctraits@N@std@@SANN@Z(double) std_Ctraits_double_tan
 @ cdecl ?tan@?$_Ctraits@O@std@@SAOO@Z(double) std_Ctraits_long_double_tan
-@ stub -arch=win32 ?tan@std@@YA?AV?$complex@M@1@ABV21@@Z
-@ stub -arch=win32 ?tan@std@@YA?AV?$complex@N@1@ABV21@@Z
-@ stub -arch=win32 ?tan@std@@YA?AV?$complex@O@1@ABV21@@Z
-@ stub -arch=win32 ?tanh@std@@YA?AV?$complex@M@1@ABV21@@Z
-@ stub -arch=win32 ?tanh@std@@YA?AV?$complex@N@1@ABV21@@Z
-@ stub -arch=win32 ?tanh@std@@YA?AV?$complex@O@1@ABV21@@Z
+@ cdecl -arch=win32 ?tan@std@@YA?AV?$complex@M@1@ABV21@@Z(ptr ptr) complex_float_tan
+@ cdecl -arch=win32 ?tan@std@@YA?AV?$complex@N@1@ABV21@@Z(ptr ptr) complex_double_tan
+@ cdecl -arch=win32 ?tan@std@@YA?AV?$complex@O@1@ABV21@@Z(ptr ptr) complex_double_tan
+@ cdecl -arch=win32 ?tanh@std@@YA?AV?$complex@M@1@ABV21@@Z(ptr ptr) complex_float_tanh
+@ cdecl -arch=win32 ?tanh@std@@YA?AV?$complex@N@1@ABV21@@Z(ptr ptr) complex_double_tanh
+@ cdecl -arch=win32 ?tanh@std@@YA?AV?$complex@O@1@ABV21@@Z(ptr ptr) complex_double_tanh
 @ cdecl -arch=arm ?tellg@?$basic_istream@DU?$char_traits@D@std@@@std@@QAA?AV?$fpos@H@2@XZ(ptr ptr) basic_istream_char_tellg
 @ thiscall -arch=i386 ?tellg@?$basic_istream@DU?$char_traits@D@std@@@std@@QAE?AV?$fpos@H@2@XZ(ptr ptr) basic_istream_char_tellg
 @ cdecl -arch=win64 ?tellg@?$basic_istream@DU?$char_traits@D@std@@@std@@QEAA?AV?$fpos@H@2@XZ(ptr ptr) basic_istream_char_tellg
diff --git a/dlls/msvcp71/msvcp71.spec b/dlls/msvcp71/msvcp71.spec
index b7a5234..967f914 100644
--- a/dlls/msvcp71/msvcp71.spec
+++ b/dlls/msvcp71/msvcp71.spec
@@ -8,8 +8,8 @@
 @ cdecl -arch=win64 ??$?5DU?$char_traits@D@std@@@std@@YAAEAV?$basic_istream@DU?$char_traits@D@std@@@0@AEAV10@PEAC@Z(ptr ptr) basic_istream_char_read_ch
 @ cdecl -arch=win32 ??$?5DU?$char_traits@D@std@@@std@@YAAAV?$basic_istream@DU?$char_traits@D@std@@@0@AAV10@PAD@Z(ptr ptr) basic_istream_char_read_str
 @ cdecl -arch=win64 ??$?5DU?$char_traits@D@std@@@std@@YAAEAV?$basic_istream@DU?$char_traits@D@std@@@0@AEAV10@PEAD@Z(ptr ptr) basic_istream_char_read_str
-@ stub -arch=win32 ??$?5DU?$char_traits@D@std@@@std@@YAAAV?$basic_istream@DU?$char_traits@D@std@@@0@AAV10@PAE@Z
-@ stub -arch=win64 ??$?5DU?$char_traits@D@std@@@std@@YAAEAV?$basic_istream@DU?$char_traits@D@std@@@0@AEAV10@PEAE@Z
+@ cdecl -arch=win32 ??$?5DU?$char_traits@D@std@@@std@@YAAAV?$basic_istream@DU?$char_traits@D@std@@@0@AAV10@PAE@Z(ptr ptr) basic_istream_char_read_str
+@ cdecl -arch=win64 ??$?5DU?$char_traits@D@std@@@std@@YAAEAV?$basic_istream@DU?$char_traits@D@std@@@0@AEAV10@PEAE@Z(ptr ptr) basic_istream_char_read_str
 @ cdecl -arch=win32 ??$?5DU?$char_traits@D@std@@V?$allocator@D@1@@std@@YAAAV?$basic_istream@DU?$char_traits@D@std@@@0@AAV10@AAV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@0@@Z(ptr ptr) basic_istream_char_read_bstr
 @ cdecl -arch=win64 ??$?5DU?$char_traits@D@std@@V?$allocator@D@1@@std@@YAAEAV?$basic_istream@DU?$char_traits@D@std@@@0@AEAV10@AEAV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@0@@Z(ptr ptr) basic_istream_char_read_bstr
 @ cdecl -arch=win32 ??$?5GU?$char_traits@G@std@@@std@@YAAAV?$basic_istream@GU?$char_traits@G@std@@@0@AAV10@AAG@Z(ptr ptr) basic_istream_wchar_read_ch
@@ -1798,8 +1798,8 @@
 @ thiscall -arch=i386 ??5?$basic_istream@GU?$char_traits@G@std@@@std@@QAEAAV01@AAF@Z(ptr ptr) basic_istream_short_read_short
 @ cdecl -arch=win64 ??5?$basic_istream@GU?$char_traits@G@std@@@std@@QEAAAEAV01@AEAF@Z(ptr ptr) basic_istream_short_read_short
 @ stub -arch=arm ??5?$basic_istream@GU?$char_traits@G@std@@@std@@QAAAAV01@AAG@Z
-@ stub -arch=i386 ??5?$basic_istream@GU?$char_traits@G@std@@@std@@QAEAAV01@AAG@Z
-@ stub -arch=win64 ??5?$basic_istream@GU?$char_traits@G@std@@@std@@QEAAAEAV01@AEAG@Z
+@ thiscall -arch=i386 ??5?$basic_istream@GU?$char_traits@G@std@@@std@@QAEAAV01@AAG@Z(ptr ptr) basic_istream_wchar_read_ushort
+@ cdecl -arch=win64 ??5?$basic_istream@GU?$char_traits@G@std@@@std@@QEAAAEAV01@AEAG@Z(ptr ptr) basic_istream_wchar_read_ushort
 @ cdecl -arch=arm ??5?$basic_istream@GU?$char_traits@G@std@@@std@@QAAAAV01@AAH@Z(ptr ptr) basic_istream_short_read_int
 @ thiscall -arch=i386 ??5?$basic_istream@GU?$char_traits@G@std@@@std@@QAEAAV01@AAH@Z(ptr ptr) basic_istream_short_read_int
 @ cdecl -arch=win64 ??5?$basic_istream@GU?$char_traits@G@std@@@std@@QEAAAEAV01@AEAH@Z(ptr ptr) basic_istream_short_read_int
diff --git a/dlls/msvcp80/msvcp80.spec b/dlls/msvcp80/msvcp80.spec
index b06086f..74f9e30 100644
--- a/dlls/msvcp80/msvcp80.spec
+++ b/dlls/msvcp80/msvcp80.spec
@@ -2010,8 +2010,8 @@
 @ thiscall -arch=i386 ??5?$basic_istream@GU?$char_traits@G@std@@@std@@QAEAAV01@AAF@Z(ptr ptr) basic_istream_short_read_short
 @ cdecl -arch=win64 ??5?$basic_istream@GU?$char_traits@G@std@@@std@@QEAAAEAV01@AEAF@Z(ptr ptr) basic_istream_short_read_short
 @ stub -arch=arm ??5?$basic_istream@GU?$char_traits@G@std@@@std@@QAAAAV01@AAG@Z
-@ stub -arch=i386 ??5?$basic_istream@GU?$char_traits@G@std@@@std@@QAEAAV01@AAG@Z
-@ stub -arch=win64 ??5?$basic_istream@GU?$char_traits@G@std@@@std@@QEAAAEAV01@AEAG@Z
+@ thiscall -arch=i386 ??5?$basic_istream@GU?$char_traits@G@std@@@std@@QAEAAV01@AAG@Z(ptr ptr) basic_istream_wchar_read_ushort
+@ cdecl -arch=win64 ??5?$basic_istream@GU?$char_traits@G@std@@@std@@QEAAAEAV01@AEAG@Z(ptr ptr) basic_istream_wchar_read_ushort
 @ cdecl -arch=arm ??5?$basic_istream@GU?$char_traits@G@std@@@std@@QAAAAV01@AAH@Z(ptr ptr) basic_istream_short_read_int
 @ thiscall -arch=i386 ??5?$basic_istream@GU?$char_traits@G@std@@@std@@QAEAAV01@AAH@Z(ptr ptr) basic_istream_short_read_int
 @ cdecl -arch=win64 ??5?$basic_istream@GU?$char_traits@G@std@@@std@@QEAAAEAV01@AEAH@Z(ptr ptr) basic_istream_short_read_int
